File: GFG POTD/032-February-2026/015-chocolate-distribution-problem.java
// https://www.geeksforgeeks.org/problems/chocolate-distribution-problem3825/1/




import java.util.*;

class Solution {
    public int findMinDiff(ArrayList<Integer> arr, int m) {
        Collections.sort(arr);
        int i = 0, j = m - 1;
        int result = Integer.MAX_VALUE;
        while (j < arr.size()) {
            result = Math.min(result, arr.get(j) - arr.get(i));
            i ++;
            j ++;
        }
        return result;
    }
}

File: GFG POTD/032-February-2026/015-chocolate-distribution-problem.py
# https://www.geeksforgeeks.org/problems/chocolate-distribution-problem3825/1/



class Solution:

    def findMinDiff(self, arr,M):
        result = float('Inf')
        arr.sort()
        i, j = 0, M - 1

        while j < len(arr):
            result = min(result, arr[j] - arr[i])
            i += 1
            j += 1

        return result